flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/stopwithsavepoint/StopWithSavepointOperationHandlerImpl.java

```diff
@@ -236,10 +260,17 @@ public State onExecutionsFinished() {
         @Override
         public State onAnyExecutionNotFinished(
                 Iterable<ExecutionState> notFinishedExecutionStates) {
-            terminateExceptionallyWithGlobalFailover(
+            terminateExceptionallyWithGlobalFailoverOnUnfinishedExecution(
                     notFinishedExecutionStates, completedSavepoint.getExternalPointer());
             return new FinalState();
         }
+
+        @Override
+        public State onAbort(Throwable cause) {
+            terminateExceptionallyWithGlobalFailover(
+                    cause, completedSavepoint.getExternalPointer());
```

Comment:
Why do we fail globally `onAbort`? Wouldn't it make more sense to say that `onAbort` stops the stop with savepoint operation but the caller has to make sure that the system continues correctly?